### ISSUE TITLE
Handle dependency failures in Linux when attempting to run Cypress

### ISSUE THREAD

**Issue by brian-mann** [cypress-io/cypress#317]
Related to #315 and #316 and #457.

There is a longstanding issue with Cypress and it failing to run when there are missing system dependencies in Linux.

The short answer is that the CLI tool automatically ignores `stderr` because Chromium is an unfathomably noisy process and there is no way to tell dependency errors apart from all the normal crap that is logged. This gives the appearance that Cypress just "hangs". It's not really hanging, in fact it could not run.

**Workaround**
To check to see if this is happening just invoke the Cypress binary directly after installing:

```bash
## install cypress
cypress install

## invoke binary directly
/home/YOUR_USER/.cypress/Cypress/Cypress
```

If there are dependency errors they will be immediately logged out to you. Just `sudo apt-get install` them.

**Actual Solution**
The solution for this is very do-able, and has been done by other types of applications in our same situation.

1. Whenever we detect that a "new" version of Cypress has been installed but has not been run the first time we attempt to launch the application.
2. While we do this, we capture `stderr` and if the output fails then we know the system is missing dependencies.
3. We then display the stored `stderr` which hopefully only displays dependency problems. Alternatively as long as we know what dependencies are required ahead of time for each OS we could list them all out or possibly even figure out whether the system has them installed ahead of time. We could also tell the user exactly what to do, like: `sudo apt-get install dep1 dep2 dep3`
4. Once the application runs on the "new" version, we cache this somewhere and don't do this check again (it is quite time consuming).

**Comment by RandallKent** [comment]
```
[5550:0113/135853:FATAL:platform_font_linux.cc(63)] Check failed: typeface. Could not find any font: 0x7fff60f2c808, sans
#0 0x000001e52dce <unknown>
#1 0x000001e68acb <unknown>
#2 0x000001937c3a <unknown>
#3 0x0000019378d6 <unknown>
#4 0x000001938556 <unknown>
#5 0x00000192acd9 <unknown>
#6 0x00000192d525 <unknown>
#7 0x00000192d399 <unknown>
#8 0x000002631ad6 <unknown>
#9 0x000002631844 <unknown>
#10 0x00000339f6a1 <unknown>
#11 0x0000035b77af brightray::InspectableWebContentsViewViews::InspectableWebContentsViewViews()
#12 0x0000035b761f brightray::CreateInspectableContentsView()
#13 0x0000035a94e0 brightray::InspectableWebContentsImpl::InspectableWebContentsImpl()
#14 0x0000035a8b3f brightray::InspectableWebContents::Create()
#15 0x00000351c8c8 atom::CommonWebContentsDelegate::InitWithWebContents()
#16 0x0000034f06d6 atom::api::WebContents::InitWithSessionAndOptions()
#17 0x0000034f0f5f atom::api::WebContents::WebContents()
#18 0x0000034f9dc6 atom::api::WebContents::Create()
#19 0x00000350a495 atom::api::Window::Window()
#20 0x00000350bc7e atom::api::Window::New()
#21 0x0000034cd56c mate::internal::InvokeNew<>()
#22 0x0000034cd6ff mate::internal::Dispatcher<>::DispatchToCallback()
#23 0x7fa2ee805868 v8::internal::FunctionCallbackArguments::Call()
#24 0x7fa2ee86377c <unknown>
#25 0x3782ffb06147 <unknown>

/usr/bin/xvfb-run: line 181:  5550 Aborted                 DISPLAY=:$SERVERNUM XAUTHORITY=$AUTHFILE "$@" 2>&1
```


Resolved on CentOS 7 with `yum install gnu-free-sans-fonts*`

**Comment by bahmutov** [comment]
We now handle missing XVFB / one of Linux dependencies better. Here are the messages you will see in CI in each case

## Missing XVFB

```
⧖ Verifying Cypress executable...
----------
Looks like your system is missing a must have dependency: XVFB
Install XVFB and run Cypress again.
Our CI documentation provides more information how to configure dependencies

https://on.cypress.io/required-dependencies
----------

Caught error trying to run XVFB: "spawn Xvfb ENOENT"

----------
Platform: linux
Version: Debian - 8.8

✖ Failed to verify Cypress executable
```

## XVFB is present, but one of libraries is missing

```
⧖ Verifying Cypress executable...
----------
Problem running Cypress application
This is usually caused by a missing library or dependency.
The error below should indicate which dependency is missing.
Read our doc on CI dependencies for more information:

https://on.cypress.io/required-dependencies
----------

/home/person/app/node_modules/cypress/dist/Cypress/Cypress: error while loading shared libraries: libgtk-x11-2.0.so.0: cannot open shared object file: No such file or directory

----------
Platform: linux
Version: Debian - 8.8

✖ Failed to verify Cypress executable.
```

## Everything is fine

```
⧖ Verifying Cypress executable...
✓ Successfully verified Cypress executable
```

## Binary version is different from `package.json` version

Because we now install Cypress binary using NPM dependency, the binary version might not match version specified in the user's `package.json`. In that case we show a warning message

```
⧖ Verifying Cypress executable...
! Installed version (0.19.4) does not match package version (0.20.0)
```

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
